sway/desktop/render.c

```diff
@@ -1379,10 +1679,48 @@ static void render_seatops(struct sway_output *output,
 	}
 }
 
+static bool find_blurred_con_iterator(struct sway_container *con, void *data) {
+	struct sway_view *view = con->view;
+	if (!view || !con->blur_enabled) {
+		return false;
+	}
+	// Only test floating windows when xray is enabled
+	if (container_is_floating(con) && !config->blur_xray) {
+		return false;
+	}
+	return !view->surface->opaque;
+}
+
+static void should_blur_draw_optimize(struct sway_output *sway_output,
+		pixman_region32_t *damage) {
+	struct fx_renderer *renderer = sway_output->renderer;
+	struct wlr_output *output = sway_output->wlr_output;
+	struct sway_workspace *workspace = sway_output->current.active_workspace;
+
+	if (!workspace_is_visible(workspace) || !renderer->blur_buffer_dirty) {
+		return;
+	}
+
+	if (!workspace_find_container(workspace, find_blurred_con_iterator, NULL)) {
+		renderer->blur_optimize_should_render = false;
+		return;
+	}
+
+	// Damage the whole output
+	int width, height;
+	wlr_output_transformed_resolution(output, &width, &height);
+	pixman_region32_union_rect(damage, damage, 0, 0, width, height);
+
+	renderer->blur_optimize_should_render = true;
+}
```

Comment:
let's convert this to a boolean function (as the name would suggest), and move the damage tracking to `output_render` after this is called

Reply:
Fixed in [92daace](https://github.com/WillPower3309/swayfx/pull/120/commits/92daaced56e4b23be0e609e4bf4fb828e52c8910)